Source/IGListManagementPerformable.h

```diff
@@ -0,0 +1,270 @@
+
+/**
+ Executes a group of updates.
+
+ @param updates A block containing all of the updates to batch
+ @param completion A block that will be called once all updates have been executed. The `BOOL` parameter indicates if any associated animations have completed succesfully: `YES` for success and `NO` if the animations were interrupted. May be `nil`.
+ */
+- (void)performBatchUpdates:(void (^)(void))updates
+                 completion:(void (^ __nullable)(BOOL))completion;
+
+/**
+ Perform an immediate reload of the data in the data source, discarding the old objects.
+ */
+- (void)reloadData;
+
+/**
+ Reload the data only for the specific sections.
+
+ @param sections Indexes of sections to reload.
+ */
+- (void)reloadSections:(NSIndexSet *)sections;
+
+/**
+ Insert new sections at the specified indexes.
+
+ @param sections Indexes of the sections to insert.
+ */
+- (void)insertSections:(NSIndexSet *)sections;
+
+/**
+ Delete existing sections at the specified indexes.
+
+ @param sections Indexes of the existing sections to delete.
+ */
+- (void)deleteSections:(NSIndexSet *)sections;
+
+/**
+ Move an existing section from one index to another final index.
+
+ @param section Initial index for the existing section.
+ @param newSection Final index for the existing section.
+ */
+- (void)moveSection:(NSInteger)section toSection:(NSInteger)newSection;
+
+/**
+ Indicate that new data is being added to the data source at specific index paths. This should cause any corresponding view(s) to be created.
+
+ @param indexPaths The index paths that have newly added data.
+ */
+- (void)insertItemsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths;
+
+/**
+ Reload existing data from the data source and redisplay the associated view(s).
+
+ @param indexPaths The index paths of the existing objects that should be reloaded.
+ */
+- (void)reloadItemsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths;
+
+/**
+ Indicate that existing data is being removed from the data source at specific index paths. This should cause the corresponding view(s) to be removed also.
+
+ @param indexPaths The index paths of the objects being removed.
+ */
+- (void)deleteItemsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths;
+
+/**
+ Indicate that an existing object is moving from one index path to another. This should cause any corresponding view(s) to move also.
+
+ @param indexPath The initial index path of the object.
+ @param newIndexPath The final index path of the object.
+ */
+- (void)moveItemAtIndexPath:(NSIndexPath *)indexPath toIndexPath:(NSIndexPath *)newIndexPath;
+
+/**
+ The number of sections being represented.
+ */
+- (NSInteger)numberOfSections;
+
+/**
+ The number of items being represented in a given section.
+
+ @param section The section index being queried.
+ */
+- (NSInteger)numberOfItemsInSection:(NSInteger)section;
+
+/**
+ The visible views of the list being represented.
+ */
+- (NSArray<__kindof UIView *> *)visibleCells;
```

Comment:
replied above. I think the best we can do is a new protocol, `IGListReusableView`.

basically, this:

https://github.com/jessesquires/JSQDataSourcesKit/blob/develop/Source/ViewFactory.swift#L95-L132